### PR TITLE
Add model for T2w lumbar SC segmentation

### PR DESCRIPTION
**Pull request by ASevag**
#### GitHub

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [ ] I've linked relevant issues in the PR body
- [x] I've applied [the relevant labels](https://intranet.neuro.polymtl.ca/geek-tips/contributing#pr-labels-a-href-pr-labels-id-pr-labels-a) to this PR
- [x] I've applied a [release milestone](https://github.com/spinalcordtoolbox/spinalcordtoolbox/milestones) (major, minor, patch) in line with [Semantic Versioning guidelines](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Misc%3A-Creating-a-new-release#convention-for-naming-releases) 
- [x] I've assigned a reviewer

#### PR contents

- [x] I've consulted [SCT's internal developer documentation](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki) to ensure my contribution is in line with any relevant design decisions
- [ ] I've added [relevant tests](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Programming%3A-Tests) for my contribution
- [ ] I've updated the [relevant documentation](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Programming%3A-Documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

## Description
Added the deepseg model and task for lumbar SC segmentation with T2w contrast from data sent by EPFL.

### CODE REVIEW

File: spinalcordtoolbox/deepseg/models.py [hunk 1]
```diff
@@ -103,6 +103,14 @@
         "description": "SC/GM multiclass segmentation on T2*-w contrast at 7T. The model was created by N.J. Laines Medina, V. Callot and A. Le Troter at CRMBM-CEMEREM Aix-Marseille University",
         "contrasts": ["t2star"],
         "default": False,
+    },
+     "model_seg_epfl_t2w_lumbar_sc": {
```
Comment (joshuacwnewton) [inline spinalcordtoolbox/deepseg/models.py:107]: This change will help to fix one of the errors from the [failed linting check](https://github.com/spinalcordtoolbox/spinalcordtoolbox/runs/6011424804?check_suite_focus=true):

```suggestion
    },
    "model_seg_epfl_t2w_lumbar_sc": {
```

File: spinalcordtoolbox/deepseg/models.py [hunk 2]
```diff
@@ -136,7 +144,10 @@
          'models': ['model_seg_exvivo_gm-wm_t2_unet2d-multichannel-softseg']},
     'seg_gm_sc_7t_t2star':
         {'description': 'SC/GM seg on T2*-weighted at 7T.',
-         'models': ['model_7t_multiclass_gm_sc_unet2d']}
+         'models': ['model_7t_multiclass_gm_sc_unet2d']},
+    'seg_lumbar_sc_t2w':
+        {'description': 'Lumbar SC segmentation on T2w contrast with 3D UNet.',
+        'models': ['model_seg_epfl_t2w_lumbar_sc']}
```
Comment (joshuacwnewton) [inline spinalcordtoolbox/deepseg/models.py:150]: This change will help to fix one of the errors from the [failed linting check](https://github.com/spinalcordtoolbox/spinalcordtoolbox/runs/6011424804?check_suite_focus=true):

```suggestion
        {'description': 'Lumbar SC segmentation on T2w contrast with 3D UNet.',
         'models': ['model_seg_epfl_t2w_lumbar_sc']}
```

File: spinalcordtoolbox/deepseg/models.py [hunk 3]
```diff
@@ -103,6 +103,14 @@
...
         "contrasts": ["t2star"],
         "default": False,
+    },
+     "model_seg_epfl_t2w_lumbar_sc": {
+        "url": [
+            "https://github.com/ivadomed/lumbar_seg_EPFL/releases/download/r20220411/model_seg_epfl_t2w_lumbar_sc.zip"
+        ],
+        "description": "Lumbar SC segmentation on T2w contrast with 3D UNet",
+        "contrasts": ["t2"],
+        "default": False,
```
Comment (joshuacwnewton) [inline spinalcordtoolbox/deepseg/models.py:113]: Just as a quick heads-up: Right now, there is a [separate PR open](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3541) that will add 2 new fields to each model/task:

* `url`: The URL for the project's page (e.g. a repo for the training source code, if you have one)
* `long_description`: A more in-depth description that summarizes the methods used to train the model (e.g. training datasets, architecture, etc.)

So, could you please write up that information for me? You don't have to add it to this PR or anything -- you can just share it in a reply, and then I will add it to the other PR #3541 later on. :) 

You can see examples of what these fields are supposed to look like in: https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3541/files

**Review by joshuacwnewton — CHANGES_REQUESTED** [review]
Thank you so much for adding this model! ❤️ 🎉

File: spinalcordtoolbox/deepseg/models.py (hunk 3 above)
Comment (ASevag) [reply spinalcordtoolbox/deepseg/models.py:113]: Sure.
Repo URL: https://github.com/ivadomed/lumbar_seg_EPFL

long_description:
This segmentation model for T2w spinal cord segmentation uses a 3D UNet architecture, and was created with the `ivadomed` package. Training data was provided by Nawal Kinany and Dimitry Van De Ville of EPFL, with the files consisting of T2w scans (and manual spinal cord segmentations) of 11 patients.

Thanks again!

Comment (ASevag) [reply spinalcordtoolbox/deepseg/models.py:113]: No, there isn't. They were all healthy AFAIK.

Comment (joshuacwnewton) [reply spinalcordtoolbox/deepseg/models.py:113]: Thank you! That's perfect, then. :)

**Review by joshuacwnewton — APPROVED** [review]

**Comment by joshuacwnewton** [comment]
Thank you very much for promptly responding to my change requests! Also, I appreciate the work you did on this project, and I'm very happy to see the end results available in SCT. ❤️

I'm going to merge this now, but one last note: The SCT v5.6 release will happen at the end of April, at which point this feature will be advertised to SCT's userbase. :)

**Pull request merged**